docs/reference/troubleshooting/troubleshooting-unbalanced-cluster.asciidoc

```diff
@@ -0,0 +1,73 @@
+as long as they are assigned with the shards with greater ingest load forecast.
+
+Use <<cat-allocation,cat allocation command>> to list workloads per node:
+
+[source,console]
+--------------------------------------------------
+GET /_cat/allocation?v
+--------------------------------------------------
+// TEST[s/^/PUT test\n{"settings": {"number_of_replicas": 0}}\n/]
+
+The API returns the following response:
+
+[source,text]
+--------------------------------------------------
+shards shards.undesired write_load.forecast disk.indices.forecast disk.indices disk.used disk.avail disk.total disk.percent host      ip        node    node.role
+     1                0                 0.0                  260b         260b    47.3gb     43.4gb    100.7gb           46 127.0.0.1 127.0.0.1 CSUXak2 himrst
+--------------------------------------------------
+// TESTRESPONSE[s/\d+(\.\d+)?[tgmk]?b/\\d+(\\.\\d+)?[tgmk]?b/ s/46/\\d+/]
+// TESTRESPONSE[s/CSUXak2 himrst/.+/ non_json]
+
+This response contains following components that influence the balancing
+
+* `shards` is the current number of shards allocated to the node
+* `shards.undesired` is the number of shards that needs to be moved to other nodes to finish balancing
+* `disk.indices.forecast` expected disk usage according to the projected shard growth
+* `write_load.forecast` project ingest load associated with this node
+
+A cluster is considered balanced when no shard movements are scheduled (all `shards.undesired` are equal to 0).
+At this state it should have spread workload reasonably (eg. some nodes might have fewer shards but more ingest load).
+
+Some operations such as node restarting or decommissioning or changing cluster allocation settings
+are disruptive and might require multiple shards to move or rebalance.
+
+Shard movement order is not deterministic and mostly determined by the source and target node readiness to move a shard.
+While rebalancing is in progress some nodes might appear busier then others.
+
+When shard is allocated to undesired node it utilizes the resources of the current node instead of target.
+This might cause a hotspot (disk or CPU) when multiple shards are residing on the current node that have not been
+moved to their corresponding targets yet.
+
+If cluster is taking a long time to finish rebalancing you might find following log entries:
+[source,text]
+--------------------------------------------------
+[WARN][o.e.c.r.a.a.DesiredBalanceReconciler] [10%] of assigned shards (10/100) are not on their desired nodes, which exceeds the warn threshold of [10%]
+--------------------------------------------------
+This is not concerning as long as the number of such shards is decreasing and this warning appears occasionally,
+for example after rolling restarts or changing allocation settings.
+
+If cluster is experiencing this warning for extended period of time, it is possible that the desired balance diverged
```

Comment:
```suggestion
If the cluster has this warning for an extended period of time, it is possible that the desired balance has diverged
```

Reply:
Maybe we could be more quantitative than "an extended period of time" here?

Reply:
I would not mind, but I am not sure how to quantify it as it depends on many things such as cluster size, amount/size of shards, background load, etc